atLeast = 0 not working if mocked method never gets called

### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

verfiy with atLeast = 0 should pass even if the mocked function is never called

### Current Behavior

Only passes if the mocked function got called at least once with different parameters

### Failure Information (for bugs)

#### Stack trace

```
verification failed: call 1 of 1: MockCls(#1).op(eq(0))) was not called
java.lang.AssertionError: Verification failed: call 1 of 1: MockCls(#1).op(eq(0))) was not called
	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:63)
	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:64)
	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalVerify(API.kt:120)
	at io.mockk.MockKKt.verify(MockK.kt:218)
	at io.mockk.MockKKt.verify$default(MockK.kt:209)
	at io.mockk.it.VerifyAtLeastAtMostExactlyTest.atLeastNeverAtAll(VerifyAtLeastAtMostExactlyTest.kt:200)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
```

#### Minimal reproducible code (the gist of this issue)

```kotlin
package io.mockk.it

import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import kotlin.test.Test

class VerifyAtLeastAtMostExactlyTest {

    val mock = mockk<MockCls>()

    @Test
    fun atLeastNeverAtAll() {
        every { mock.op(0) } returns 1

        verify(atLeast = 0) { mock.op(0) }
    }

    class MockCls {
        fun op(a: Int) = a + 1
    }
}
```

--- a/modules/mockk/src/commonMain/kotlin/io/mockk/impl/verify/UnorderedCallVerifier.kt
+++ b/modules/mockk/src/commonMain/kotlin/io/mockk/impl/verify/UnorderedCallVerifier.kt
@@ -82,7 +82,9 @@ open class UnorderedCallVerifier(
             }
         } else when (allCallsForMockMethod.size) {
             0 -> {
-                if (allCallsForMock.isEmpty()) {
+                if(min == 0) {
+                    VerificationResult.OK(listOf())
+                } else if (allCallsForMock.isEmpty()) {
                     VerificationResult.Failure("$callIdxMsg was not called")
                 } else {
                     VerificationResult.Failure(safeToString.exec {

--- a/modules/mockk/src/commonTest/kotlin/io/mockk/it/VerifyAtLeastAtMostExactlyTest.kt
+++ b/modules/mockk/src/commonTest/kotlin/io/mockk/it/VerifyAtLeastAtMostExactlyTest.kt
@@ -194,6 +194,13 @@ class VerifyAtLeastAtMostExactlyTest {
     }
 
     @Test
+    fun atLeastNeverAtAll() {
+        every { mock.op(0) } returns 1
+
+        verify(atLeast = 0) { mock.op(0) }
+    }
+
+    @Test
     fun atLeastNever() {
         every { mock.op(0) } returns 1
         mock.op(0)

